tests/unit/test_tools_archive.py

```diff
@@ -0,0 +1,171 @@
+    assert expected_location == location
+    # Test it has written the correct content
+    assert expected_archived == location.read_text()
+
+
+def test_issue_has_no_comments_default(issue_100):
+    """Test that an issue has no comments per default."""
+    assert not issue_100.has_comments()
+
+
+def test_issue_has_comments(issue_1470):
+    """Test that an issue has comments."""
+    assert issue_1470.has_comments()
+    assert issue_1470.comments_number == 41
+
+
+def test_comments_fetch(mocker, issue_1470):
+    """Test the comments fetching.
+
+    return a list of dictionary, each dictionary represents a comment.
+    """
+    # Before fetching the comments
+    assert type(issue_1470.comments) == list
+    assert len(issue_1470.comments) == 0
+    # Instead of fetching from the network, we get a fixture
+    fake_json = mocker.patch.object(model.Issue, 'recursive_fetch')
+    fake_json.return_value = json.loads(get_fixture('1470-comments-all.json'))
+    issue_1470.fetch_comments()
+    # actual tests after fetching
+    assert len(issue_1470.comments) == 41
+    assert type(issue_1470.comments) == list
+    assert type(issue_1470.comments[0]) == model.Comment
+    comment_0 = issue_1470.comments[0]
+    assert comment_0.author == 'karlcow'
+    assert 'ok. Thanks a lot' in comment_0.body
+    d = datetime.datetime(2015, 7, 28, 9, 25, 3, tzinfo=datetime.timezone.utc)
+    assert comment_0.created_at == d
+    assert comment_0.updated_at == d
+
+
+def test_comments_fetch_no_comments(issue_100):
+    """Test the comments fetching without comments available."""
+    # Before fetching the comments
+    issue_100.fetch_comments()
+    assert len(issue_100.comments) == 0
+
+
+def test_make_request(mocker):
+    fake_resp = mocker.patch.object(model.requests, 'get')
+    fake_resp.return_value.status_code = 200
+    actual = model.make_request('http://example.org/')
+    assert actual.status_code == 200
+
+
+def test_http_error_log_comments_fetch(mocker, caplog, issue_1470):
+    """Test that we record a log message for HTTP error when fetching."""
+    fake_r = mocker.patch.object(model.requests, 'get')
+    fake_r.side_effect = HTTPError()
+    fake_r.status_code = 400
+    caplog.set_level(logging.WARNING)
+    model.Issue.recursive_fetch('https://example.org/')
+    assert 'Fetching comments failed' in caplog.text
+
+
+@pytest.mark.skip(reason="TODO: JSON for ALL comments")
+def test_recursive_fetch(mocker):
+    """Test that we receive a JSON of all comments."""
+    # This a recursive function which is probably hard to test.
+    pass
+
+
+def test_to_datetime():
+    """convert date string to datetime object."""
+    expected_date = datetime.datetime(2020, 8, 14, 12, 34, 56,
+                                      tzinfo=datetime.timezone.utc)
+    assert model.to_datetime('2020-08-14T12:34:56Z') == expected_date
+
+
+def test_issue_is_locked(issue_100):
+    """test if an issue is locked."""
+    assert issue_100.locked
```

Comment:
Isn't this just testing that the fixture data has `"locked": true`? Is that a useful test (forgive me if I am missing something)? 

(I guess it does test that the `locked` property is set on the dataclass, but we don't really test all the props similarly (number, title, body, etc.))

Reply:
Yes it's a remaining of a strategy that I was planning to use, but I have not implemented. Some of our closed issues are not locked.